yt_dlp/extractor/nest.py

```diff
@@ -0,0 +1,114 @@
+from .common import InfoExtractor
+from ..utils import ExtractorError, float_or_none, url_or_none
+from ..utils.traversal import traverse_obj
+
+
+class NestIE(InfoExtractor):
+    _VALID_URL = r'https?://video\.nest\.com/(?:embedded/)?live/(?P<id>\w+)'
+    _EMBED_REGEX = [rf'<iframe[^>]+\bsrc=[\'"](?P<url>{_VALID_URL})']
+    _TESTS = [{
+        'url': 'https://video.nest.com/embedded/live/4fvYdSo8AX?autoplay=0',
+        'info_dict': {
+            'id': '4fvYdSo8AX',
+            'ext': 'mp4',
+            'title': 'startswith:Outside ',
+            'alt_title': 'Outside',
+            'description': '<null>',
+            'location': 'Los Angeles',
+            'availability': 'public',
+            'thumbnail': r're:https?://',
+            'live_status': 'is_live',
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+    }, {
+        'url': 'https://video.nest.com/live/4fvYdSo8AX',
+        'only_matching': True,
+    }]
+    _WEBPAGE_TESTS = [{
+        'url': 'https://www.pacificblue.biz/noyo-harbor-webcam/',
+        'info_dict': {
+            'id': '4fvYdSo8AX',
+            'ext': 'mp4',
+            'title': 'startswith:Outside ',
+            'alt_title': 'Outside',
+            'description': '<null>',
+            'location': 'Los Angeles',
+            'availability': 'public',
+            'thumbnail': r're:https?://',
+            'live_status': 'is_live',
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        data = self._download_json(
+            f'https://video.nest.com/api/dropcam/cameras.get_by_public_token?token={video_id}', video_id)
+        item = traverse_obj(data, ('items', 0, {dict}))
```

Comment:
it's fatal in the very next line if `data['items'][0]` isn't a dict, and we don't use any other part of `data`, so we might as well do this above:

```py
       item = self._download_json(
            'https://video.nest.com/api/dropcam/cameras.get_by_public_token',
            video_id, query={'token': video_id})['items'][0]
```

```suggestion
```